htlcswitch/link_test.go

```diff
@@ -277,125 +275,93 @@ func TestChannelLinkBidirectionalOneHopPayments(t *testing.T) {
 				n.aliceServer, n.bobServer, firstHop,
 				hopsForwards, amt, htlcAmt, totalTimelock,
-			).Wait(5 * time.Minute)
-			resultChan <- r
+				preimage,
+			)
+			alicePreimageChan <- preimage
+			aliceErrChan <- err
 		}(i)
 	}
 
-	for i := 0; i < count/2; i++ {
-		go func(i int) {
-			r := &result{
-				start:  time.Now(),
-				number: i,
-				sender: "bob",
-			}
+	// Wait for Alice to finish filling her commitment.
+	wg.Wait()
+	close(alicePreimageChan)
+	close(aliceErrChan)
 
-			firstHop := n.aliceChannelLink.ShortChanID()
-			_, r.err = makePayment(
-				n.bobServer, n.aliceServer, firstHop,
-				hopsBackwards, amt, htlcAmt, totalTimelock,
-			).Wait(5 * time.Minute)
-			resultChan <- r
-		}(i)
+	// Now make an additional payment from Alice to Bob, this should be
+	// canceled because the commitment in this direction is full.
+	err = <-makePayment(
+		n.aliceServer, n.bobServer, firstHop, hopsForwards, amt,
+		htlcAmt, totalTimelock,
+	).err
+	if err == nil {
+		t.Fatalf("overflow payment should have failed")
+	}
+	lerr, ok := err.(*LinkError)
+	if !ok {
+		t.Fatalf("expected LinkError, got: %T", err)
 	}
 
-	maxDelay := time.Duration(0)
-	minDelay := time.Duration(math.MaxInt64)
-	averageDelay := time.Duration(0)
-
-	// Check that alice invoice was settled and bandwidth of HTLC
-	// links was changed.
-	for i := 0; i < count; i++ {
-		select {
-		case r := <-resultChan:
-			if r.err != nil {
-				t.Fatalf("unable to make payment: %v", r.err)
-			}
-
-			delay := time.Since(r.start)
-			if delay > maxDelay {
-				maxDelay = delay
-			}
-
-			if delay < minDelay {
-				minDelay = delay
-			}
-			averageDelay += delay
-
-		case <-time.After(5 * time.Minute):
-			t.Fatalf("timeout: (%v/%v)", i+1, count)
-		}
+	msg := lerr.WireMessage()
+	if _, ok := msg.(*lnwire.FailTemporaryChannelFailure); !ok {
+		t.Fatalf("expected TemporaryChannelFailure, got: %T", msg)
 	}
 
-	// TODO(roasbeef): should instead consume async notifications from both
-	// links
-	time.Sleep(time.Second * 2)
+	// Now, settle all htlcs held by bob and clear the commitment of htlcs.
+	for preimage := range alicePreimageChan {
+		preimage := preimage
 
-	// At the end Bob and Alice balances should be the same as previous,
-	// because they sent the equal amount of money to each other.
-	if aliceBandwidthBefore != n.aliceChannelLink.Bandwidth() {
-		t.Fatalf("alice bandwidth shouldn't have changed: expected %v, got %x",
-			aliceBandwidthBefore, n.aliceChannelLink.Bandwidth())
+		err = wait.NoError(func() error {
+			return n.bobServer.registry.SettleHodlInvoice(preimage)
+		}, time.Minute)
+		if err != nil {
+			t.Fatal(err)
+		}
 	}
 
-	if bobBandwidthBefore != n.firstBobChannelLink.Bandwidth() {
-		t.Fatalf("bob bandwidth shouldn't have changed: expected %v, got %v",
-			bobBandwidthBefore, n.firstBobChannelLink.Bandwidth())
+	// Ensure that all of the payments sent by alice eventually succeed.
+	for errChan := range aliceErrChan {
+		err := <-errChan
```

Comment:
Timeout chan? Or you can ofcourse of `defer timeout(t)()` :sunglasses:

Reply:
no need because this is buffered and closed